js/scorm/wrapper.js

```diff
@@ -605,76 +607,69 @@ class ScormWrapper {
   }
 
   recordInteractionScorm12(id, response, correct, latency, type) {
-
+    const children = this.getValue('cmi.interactions._children');
+    if (this.isUnsupportedLastError() || !children.includes('id') || !this.isSupported('cmi.interactions._count')) {
+      this.logUnsupported('cmi.interactions');
+      return;
+    }
     id = id.trim();
-
     const cmiPrefix = `cmi.interactions.${this.getInteractionCount()}`;
-
     this.setValue(`${cmiPrefix}.id`, id);
-    this.setValue(`${cmiPrefix}.type`, type);
-    this.setValue(`${cmiPrefix}.student_response`, response);
-    this.setValue(`${cmiPrefix}.result`, correct ? 'correct' : 'wrong');
-    if (latency !== null && latency !== undefined) this.setValue(`${cmiPrefix}.latency`, this.convertToSCORM12Time(latency));
-    this.setValue(`${cmiPrefix}.time`, this.getCMITime());
+    children.includes('type') ? this.setValue(`${cmiPrefix}.type`, type)
+      : this.logUnsupported(`${cmiPrefix}.type`);
+    children.includes('student_response') ? this.setValue(`${cmiPrefix}.student_response`, response)
+      : this.logUnsupported(`${cmiPrefix}.student_response`);
+    children.includes('result') ? this.setValue(`${cmiPrefix}.result`, correct ? 'correct' : 'wrong')
+      : this.logUnsupported(`${cmiPrefix}.result`);
+    if (latency != null) {
+      children.includes('latency') ? this.setValue(`${cmiPrefix}.latency`, this.convertToSCORM12Time(latency))
+      : this.logUnsupported(`${cmiPrefix}.latency`);
+    }
+    children.includes('time') ? this.setValue(`${cmiPrefix}.time`, this.getCMITime())
+      : this.logUnsupported(`${cmiPrefix}.time`);
   }
 
   recordInteractionScorm2004(id, response, correct, latency, type) {
-
     id = id.trim();
-
     const cmiPrefix = `cmi.interactions.${this.getInteractionCount()}`;
-
     this.setValue(`${cmiPrefix}.id`, id);
     this.setValue(`${cmiPrefix}.type`, type);
     this.setValue(`${cmiPrefix}.learner_response`, response);
     this.setValue(`${cmiPrefix}.result`, correct ? 'correct' : 'incorrect');
-    if (latency !== null && latency !== undefined) this.setValue(`${cmiPrefix}.latency`, this.convertToSCORM2004Time(latency));
+    if (latency != null) this.setValue(`${cmiPrefix}.latency`, this.convertToSCORM2004Time(latency));
```

Comment:
Never use a loose equality operator please. Always the triple equals variety. The `!= null` for undefined and null detection is by no means obvious and isn't well used anywhere else in the code.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/null#difference_between_null_and_undefined

Reply:
Can change back, but thought this should be clear as that is how the loose equality operator works, as shown in the link provided.